Packages/com.unity.inputsystem/InputSystem/Events/CompressMouseMoveEvents.cs

```diff
@@ -0,0 +1,257 @@
+                    return;
+
+                var resultEventCount = 0;
+                var resultEventSizeInBytes = 0L;
+
+                new CompressMouseMoveEvents
+                {
+                    updateType = updateType,
+                    processUntilTimestamp = processUntilTimestamp,
+                    eventPtr = events.bufferPtr.ToPointer(),
+                    eventCount = events.eventCount,
+                    eventSizeInBytes = events.sizeInBytes,
+                    resultEventCount = &resultEventCount,
+                    resultEventSizeInBytes = &resultEventSizeInBytes
+                }.Run();
+
+                events.Shrink(resultEventCount, resultEventSizeInBytes);
+            }
+        }
+
+        public InputUpdateType updateType;
+        public double processUntilTimestamp;
+
+        [NativeDisableUnsafePtrRestriction] public InputEvent* eventPtr;
+
+        public int eventCount;
+        public long eventSizeInBytes;
+
+        // Job fields are copied, so to read results back from a job we need to write to a pointer.
+        [NativeDisableUnsafePtrRestriction] public int* resultEventCount;
+        [NativeDisableUnsafePtrRestriction] public long* resultEventSizeInBytes;
+
+        public void Execute()
+        {
+            // Write down no-op results now.
+            *resultEventCount = eventCount;
+            *resultEventSizeInBytes = eventSizeInBytes;
+
+            // Step 1: early out if any mouse event is propagated via delta state event.
+            // It's a bit involved to support compression of delta state events,
+            // and given that no backend sends mouse delta events today it's probably safe to early out.
+            // Also calculate correct event count for fixed update.
+            var currentEvent = eventPtr;
+            var eventPtrs = new NativeArray<IntPtr>(eventCount + 1, Allocator.Temp);
+            var eventToProcessCount = eventCount;
+            for (var i = 0; i < eventCount; ++i)
+            {
+                eventPtrs[i] = (IntPtr)currentEvent;
+
+                if (currentEvent->type == DeltaStateEvent.Type)
+                {
+                    var stateEvent = DeltaStateEvent.FromUnchecked(currentEvent);
+                    if (stateEvent->stateFormat == MouseState.Format)
+                        return;
+                }
+
+                currentEvent = InputEvent.GetNextInMemory(currentEvent);
+
+                if (updateType == InputUpdateType.Fixed && currentEvent->internalTime >= processUntilTimestamp)
+                {
+                    eventToProcessCount = i;
+                    break;
+                }
+            }
+            eventPtrs[eventToProcessCount] = (IntPtr)currentEvent;  // Remember the end pointer, will make life easier later.
+
+            // Step 2: compress move events in-place, mark redundant events for skipping
```

Comment:
This seems like aggregation, maybe call this step aggregation since basically just summing events which have linearly additive components without losing information bits. Compression stage seems to come later.